docs/intro-to-dune/index.md

```diff
@@ -0,0 +1,284 @@
+`src/counter` directory and run
+
+```
+make serve
+```
+
+It should open a new tab in your default browser to <a
+href="http://localhost:8080/" target="_blank" rel="noreferrer
+noopener">http://localhost:8080/</a> and display your Counter component.
+
+## Serving a default app
+
+If you run `make serve` in the root directory, you'll get an error that looks
+something like this:
+
+```
+ERROR in main
+Module not found:
+Error: Can't resolve './_build/default/src//output/src//Index.js' in '...'
+```
+
+That's because if `app` environment variable isn't provided, the value passed to
+the `--entry` option is no longer a valid path. We can define a default value
+for `app` by putting this near the top of `Makefile`:
+
+```make
+ifndef app
+	app := counter
+endif
+```
+
+Now if you run `make serve` in the root directory, it will serve the counter
+app.
+
+:::info
+Feel to delete the `src/Index.re` and `src/dune` files. You won't need them
+anymore for this or later chapters. You can also delete the `scripts` section of
+your `package.json` file.
+:::
+
+Huzzah! You created a new `dune` file to build an app and a `Makefile` to
+serve that app locally. In future chapters, we'll assume that you use the same
+directory structure for each new app you build.
+
+## Exercises
+
+<b>1.</b> Repeat the steps we did for Counter and create a separate app for
+Celsius Converter.
+
+<b>2.</b> Delete `reason-react-ppx` from `src/counter/dune`'s `melange.emit`
+stanza. What compiler errors do you get?
+
+<b>3.</b> You might have noticed that every recipe in the Makefile is prefixed
+by a tab. What happens if you replace the tab in front of the `serve` rule's
+recipe with four spaces?
+
+## Overview
+
+- Dune is the build system we use to build Melange projects
+- The `dune-project` file describes the metadata for your project, primarily:
+  - The version of Dune you're using
+  - The version of the Melange plugin for Dune you're using
+  - The name of your project
+- `dune` files describe things of interest to Dune, for example:
+  - Which directories to include and which to exclude
+  - Which directories contain code that should be transpiled to JavaScript,
+    using the `melange.emit` stanza
+- Make is a build automation tool heavily used in the OCaml world
+  - `Makefile`s contain rules which can run commands based on the target you
+    pass to the `make` command
+
+## Solutions
+
+<b>1.</b> Creating a separate app for Celsius Converter, with its own `dune`,
+`Makefile`, and `Index.re` files, should look something like
+[this](https://github.com/melange-re/melange-for-react-devs/tree/develop/src/celsius-converter-option).
+
+<b>2.</b> Deleting `reason-react-ppx` from `src/counter/dune` will result in a
+compilation error:
+
+```
+File "src/counter/Counter.re", line 5, characters 2-6:
+5 |   <div
+      ^^^^
+Error: Unbound value div
+```
+
+That's because putting `reason-react-ppx` in the `preprocess/pps` field enables
+JSX in `.re` files.
```

Comment:
That's not entirely correct. Here JSX is enabled (in all reason files is), but isn't treated for any ppx so the function `div` remains untouched and since isn't defined anywhere, it complains.

Reply:
I'm not sure I understand your explanation. Can I say something more like this?

```suggestion
That's because putting `reason-react-ppx` in the `preprocess/pps` field allows
all React element functions like `div` to be visible to your JSX code.
```

Reply:
I think what Dave meant is that by default, Reason parser already processes JSX syntax, with or without `reason-react-ppx`.

For example, this reason code:

```reason
let t = <div />;
```

is equivalent to this other:

```reason
let t = [@JSX] div(~children=[], ());
```

You can see so by running refmt in the latter ([try in playground](https://melange.re/v2.1.0/playground/?language=Reason&code=bGV0IHQgPSBbQEpTWF0gZGl2KH5jaGlsZHJlbj1bXSwgKCkpOwo%3D&live=offis)).

What the `reason-react-ppx` does is to transform expressions annotated with `@JSX` attribute into calls to `React.createElement` or `ReactDOM.createElement`.

> That's because putting `reason-react-ppx` in the `preprocess/pps` field allows all React element functions like `div` to be visible to your JSX code.

I am not sure I understand what that ^ means. I would say something like:

> That's because putting `reason-react-ppx` in the `preprocess/pps` field will transform function calls to `div` into calls to `React.createElement("div", ...)`.

Reply:
Changed in https://github.com/melange-re/melange-for-react-devs/pull/6/commits/2e9aae9bc19c4ab66c98e5702fc2b40a928eb8bb to 

> That's because putting `reason-react-ppx` in the `preprocess/pps` field will
transform function calls to `div` (which isn't defined anywhere) into calls to
`React.createElement("div", ...)`.